### ISSUE TITLE
Need better detection for fixed intersections

### ISSUE THREAD

**Issue by mitxela** [mitxela/kicad-round-tracks#4]
If multiple tracks join at a pad that's part of a footprint they can get rounded away from it.

![image](https://user-images.githubusercontent.com/19376076/154812087-2c1b57d0-a996-4043-a29c-368c53b34506.png)

I'll fix this the same way vias are handled, just need to iterate over all possible fixed intersections.

In the mean time, you can move the endpoints of the tracks slightly, so they're still within the pad but not actually touching.

**Comment by mitxela** [comment]
This is fixed now I think, I'll close this when I do the next release.

It's a bit annoying that we have to explicitly check for SMD pads. I've no idea if this would ever turn up on a real PCB, but technically there's an edge case where a surface mount pad is on the opposite side of the board to a track of the same net that you **do** want to round. Thankfully this doesn't seem to have added much overhead.

![image](https://user-images.githubusercontent.com/19376076/166115682-d125a664-db40-487e-9b8a-453fdd725d8d.png)
now correctly becomes:
![image](https://user-images.githubusercontent.com/19376076/166115668-dd20c42d-6a39-4985-adde-73aa43943d4b.png)

**Comment by mitxela** [comment]
Probably overthinking this, but...

I found on a few of my designs the tracks didn't perfectly intersect the centre of the pad. Probably because I'd moved the component slightly afterwards. Really this should only melt if the intersection is not within the pad, but there are so many varieties of pad shapes, including custom polygons, that exact checking would be too expensive.

Even with the pad shapes where we can check exactly without too much trouble (circles and rectangles) I found sometimes tracks didn't get rounded when they should do: when just one track is approaching a pad, but has a bend as it enters the pad.

Now updated it to avoid rounding if the intersection is within the bounding box of a pad, but only if the remaining track ends are _outside_ of the bounding box.

So this:
![image](https://user-images.githubusercontent.com/19376076/166345477-d27a1697-cd12-4fc6-b1f6-fc1fb8c60bb1.png)
does get rounded:
![image](https://user-images.githubusercontent.com/19376076/166345495-81c5cb6e-329d-4037-bd96-a208d0f7dfe1.png)
But this:
![image](https://user-images.githubusercontent.com/19376076/166345513-be16f9b6-5d49-4418-9d1f-3be71272b700.png)
doesn't, as it would break connection to the pad:
![image](https://user-images.githubusercontent.com/19376076/166345561-9a8c72fc-79cf-4a99-91a5-e9ecc0714f3e.png)
